common/data/src/main/java/org/thingsboard/server/common/data/device/profile/MqttDeviceProfileTransportConfiguration.java

```diff
@@ -15,17 +15,30 @@
  */
 package org.thingsboard.server.common.data.device.profile;
 
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import lombok.Data;
 import org.thingsboard.server.common.data.TransportPayloadType;
 import org.thingsboard.server.common.data.DeviceTransportType;
 
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonTypeInfo(
+        use = JsonTypeInfo.Id.NAME,
+        include = JsonTypeInfo.As.PROPERTY,
+        property = "transportPayloadType")
+@JsonSubTypes({
+        @JsonSubTypes.Type(value = MqttJsonDeviceProfileTransportConfiguration.class, name = "JSON"),
+        @JsonSubTypes.Type(value = MqttProtoDeviceProfileTransportConfiguration.class, name = "PROTOBUF")})
+@JsonDeserialize(using = MqttTransportConfigurationDeserializer.class)
 @Data
-public class MqttDeviceProfileTransportConfiguration implements DeviceProfileTransportConfiguration {
+public abstract class MqttDeviceProfileTransportConfiguration implements DeviceProfileTransportConfiguration {
```

Comment:
This is completely wrong. You are breaking the main OOP principles.

You should have only one MqttDeviceProfileTransportConfiguration (not abstract) that has a field:
TransportPayloadType
and one more variable of new TransportPayloadTypeConfiguration interface (JSON or PROTOBUF)
with two sub-classes

Reply:
Ok. This is clear.